src/librustc_mir/util/pretty.rs

```diff
@@ -555,7 +555,7 @@ fn write_mir_sig(
             write!(w, "static {}", if tcx.is_mutable_static(src.def_id()) { "mut " } else { "" })?
         }
         (_, _) if is_function => write!(w, "fn ")?,
-        (None, _) => {} // things like anon const, not an item
+        (Some(DefKind::AnonConst), _) | (None, _) => {} // things like anon const, not an item
         _ => bug!("Unexpected def kind {:?}", kind),
     }
 
```

Comment:
Oh dear I just realize why we have that double space bug.

@oli-obk `kind` ends in a space (when it makes sense to do so), but the next `write!` below starts with a space (it shouldn't).

(not relevant to this PR, just spotted it)